### PR TITLE
UX Editor button to simplify the insertion of a module on an article

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### What is this?

Another button bellow the editor that allows people to easily insert a module to their article
Once the button get their position inside tinyMCE this will look a lot better.
#### How to test

Apply this PR with patch tester
Go to discover and install the new plugin
Edit an article (fronted/backend) and make sure that the module is inserted
Also insert a style in the given input, either html5 or none, and observe the changes
#### Preview

screen shot 2015-10-11 at 11 35 32

### CODE REVIEW

**Comment by brianteeman** [comment]
Your text says insert article inside article. ??

**Comment by dgt41** [comment]
Thanks Brian,

It’s the fever effect I guess<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by brianteeman** [comment]
If the tinymce update that puts the buttons into the toolbar is accepted
then this is a great addition. If its not then it just adds more clutter in
the UI. my 2c

File: administrator/language/en-GB/en-GB.plg_editors-xtd_module.sys.ini [hunk 1]
```diff
@@ -0,0 +1,10 @@
+; Joomla! Project
+; Copyright (C) 2005 - 2015 Open Source Matters. All rights reserved.
+; License GNU General Public License version 2 or later; see LICENSE.txt, see LICENSE.php
+; Note : All ini files need to be saved as UTF-8
+
+PLG_EDITORS-XTD_MODULE="Button - Module"
+PLG_MODULE_XML_DESCRIPTION="Displays a button to make it possible to insert a module into an Article. Displays a popup allowing you to choose the module."
+
+
```
Comment (zero-24) [inline administrator/language/en-GB/en-GB.plg_editors-xtd_module.sys.ini]: Can we get that clear lines out?

File: plugins/editors-xtd/module/module.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Editors-xtd.article
```
Comment (zero-24) [inline plugins/editors-xtd/module/module.php]: I guess we should use "module" here?

File: plugins/editors-xtd/module/module.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	 *
+	 * @var    boolean
+	 * @since  3.5
+	 */
+	protected $autoloadLanguage = true;
+
+	/**
+	 * Display the button
+	 *
+	 * @return array
```
Comment (zero-24) [inline plugins/editors-xtd/module/module.php]: Can we have a @since tag here?

File: plugins/editors-xtd/module/module.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<extension version="3.4" type="plugin" group="editors-xtd" method="upgrade">
+	<name>plg_editors-xtd_module</name>
+	<author>Joomla! Project</author>
+	<creationDate>October 2015</creationDate>
+	<copyright>Copyright (C) 2005 - 2015 Open Source Matters. All rights reserved.</copyright>
+	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
+	<authorEmail>admin@joomla.org</authorEmail>
+	<authorUrl>www.joomla.org</authorUrl>
+	<version>1.0.0</version>
```
Comment (zero-24) [inline plugins/editors-xtd/module/module.xml]: Version should be 3.5.0

File: plugins/editors-xtd/module/module.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<extension version="3.4" type="plugin" group="editors-xtd" method="upgrade">
```
Comment (zero-24) [inline plugins/editors-xtd/module/module.xml]: Same here should be 3.5 ;)

**Comment by dgt41** [comment]
@brianteeman I totally agree, I wouldn’t create the button if it would be displayed under the editor<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by brianteeman** [comment]
Cool

**Comment by nikosdion** [comment]
I have tested this item :white_check_mark: successfully on fdec3eabfa23402beb9dc77ce434fa357961d406<br /><br />Works as advertised. I tried both inserting a specific module and a module position. Perfect results. Thank you, Dimitris!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by rdeutz** [comment]
I have tested this item :white_check_mark: successfully on fdec3eabfa23402beb9dc77ce434fa357961d406<br /><br />works as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by zero-24** [comment]
RTC for 3.5. Thanks. :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by infograf768** [comment]
@dgt41
This is not multilanguage aware.
Can you filter the available modules in the popup by displaying only those tagged to the current language of the article edited and ALL languages when multilang is enabled? 
A bit as we do for "forcedLanguage" but also adding ALL.

**Comment by infograf768** [comment]
For the frontend, it looks possible by adding
```
		// Modal view should return only specific language and ALL
		if (JFactory::getApplication()->input->get('layout') == 'modal')
		{
			if (JFactory::getApplication()->isSite() && JLanguageMultilang::isEnabled())
			{
				$language = JFactory::getLanguage()->getTag();
				$query->where('a.language in (' . $db->quote(JFactory::getLanguage()->getTag()) . ',' . $db->quote('*') . ')');
			}
		}
```

But for the backend, we would need to check first the language of the article, if already saved.
Any idea?

**Comment by infograf768** [comment]
Corrected code above<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by dgt41** [comment]
@infograf768 I guess we can go away with the current status for the backend? Or we provide current state for not saved articles and do the extra filtering in the other occasion

**Comment by infograf768** [comment]
Agree. Can you update PR?

**Comment by infograf768** [comment]
Also, please add the update(s) sql and the new plugin in administrator/language/en-GB/install.xml

**Comment by zero-24** [comment]
Back to pending
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by infograf768** [comment]
@dgt41 
Still missing:
>Also, please add the update(s) sql and the new plugin in administrator/language/en-GB/install.xml

**Comment by zero-24** [comment]
As well as the General install sql and the Script.php (com_admin)

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by coolman01** [comment]
I have tested this item :red_circle: unsuccessfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<br /><br />I like this - seems far more useful than injecting an article.
Yes it's already possible, but adding a style is cool, and it's definitly simpler to use for newbies.
It worked as decsribed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by coolman01** [comment]
I have tested this item :white_check_mark: successfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<br /><br />I like this - seems far more useful than injecting an article.
Yes it's already possible, but adding a style is cool, and it's definitly simpler to use for newbies.
It worked as decsribed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by pascalelocher** [comment]
I have tested this item :white_check_mark: successfully on ec43648addd05d79fb8a08f9827f7038e9177b8f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/8064">issues.joomla.org/joomla-cms/8064</a>.</sub>

**Comment by Bakual** [comment]
Set to RTC. Leaving for @roland-d to merge into `3.5-dev` :smile:

**Comment by roland-d** [comment]
Thanks everybody, merged into 3.5-dev with commit  06f0d96 

Another hair pulling merge :P

**Pull request closed without merging**